src/components/SqlModal.svelte

```diff
@@ -28,6 +22,142 @@
       }, 2000);
     }
   }
+
+  function getGlamSql() {
+    const columns = [
+      'os',
+      'app_version',
+      'app_build_id',
+      'channel',
+      'metric',
+      'metric_type',
+      'key',
+      'process',
+      'client_agg_type',
+      'agg_type',
+      'total_users',
+      'mozfun.glam.histogram_cast_json(aggregates) AS aggregates',
+    ];
+    const sql = [
+      'SELECT',
+      `  ${columns.join(',\n  ')}`,
+      'FROM',
+      '  `moz-fx-data-shared-prod.telemetry.client_probe_counts`',
+      'WHERE',
+      `  metric="${$store.probe.name}"`,
+      `  AND channel="${$store.productDimensions.channel}"`,
+      $store.productDimensions.os === '*'
+        ? '  AND os IS NULL'
+        : `  AND os="${$store.productDimensions.os}"`,
+      `  AND process="${$store.productDimensions.process}"`,
+      $store.productDimensions.aggregationLevel === 'build_id'
+        ? '  AND app_build_id IS NOT NULL'
+        : '  AND app_build_id IS NULL',
+    ];
+    return sql.join('\n');
+  }
+
+  function getTelemetrySql() {
+    const { process } = $store.productDimensions;
+    let telemetryPath;
+    if ($store.probe.type === 'histogram') {
+      if (['main', 'parent'].includes(process)) {
+        telemetryPath = `payload.histograms.${$store.probe.name}`;
+      } else {
+        telemetryPath = `payload.processes.${process}.histograms.${$store.probe.name}`;
+      }
+    } else if ($store.probe.type === 'scalar') {
+      if (process === 'main') {
+        telemetryPath = `payload.processes.parent.scalars.${$store.probe.name}`;
+      } else {
+        telemetryPath = `payload.processes.${process}.scalars.${$store.probe.name}`;
+      }
+    }
+    const sql = [
+      'WITH per_build_client_day AS (',
+      '  SELECT',
+      '    PARSE_DATETIME("%Y%m%d%H%M%S", application.build_id) AS build_id,',
+      '    client_id,',
+      '    udf.histogram_normalize(',
+      '      udf.histogram_merge(',
+      '        ARRAY_AGG(',
+      '          udf.json_extract_histogram(',
+      `            ${telemetryPath}`,
+      '          )',
+      '        )',
+      '      )',
+      `  ) AS ${$store.probe.name}`,
+      '  FROM',
+      '    telemetry.main_1pct',
```

Comment:
I think we should use `telemetry.main_nightly` if the channel is nightly, 1pct is very sparse for nightly leading to weird results.

Can you also add a SQL comment somewhere with a link to the docs we have on these tables:

https://docs.telemetry.mozilla.org/datasets/main_ping_tables.html

Reply:
Great idea. I can add one at the top of the SQL.